modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/ObjectEntry1toMObjectRelatedModelsProviderImpl.java

```diff
@@ -116,23 +123,35 @@ public void disassociateRelatedModels(
 			long primaryKey2)
 		throws PortalException {
 
-		_objectEntryService.updateObjectEntry(
-			primaryKey2,
+		ObjectRelationship objectRelationship =
+			_objectRelationshipLocalService.getObjectRelationship(
+				objectRelationshipId);
+
+		ObjectDefinition systemObjectDefinition =
```

Comment:
You don't know if this is system or not until the `if` at the bottom. I would not call it `systemObjectDefinition`, maybe just `objectDefinition` or `objectDefinition2`.